crates/bevy_render/src/mesh/allocator.rs

```diff
@@ -540,12 +541,17 @@ impl MeshAllocator {
 
                 let slot_size = general_slab.element_layout.slot_size();
 
-                // Write the data in.
-                render_queue.write_buffer(
-                    buffer,
-                    allocated_range.allocation.offset as u64 * slot_size,
-                    &pad_to_alignment(data, slot_size as usize),
-                );
+                if let Some(size) = BufferSize::new((len as u64).next_multiple_of(slot_size)) {
+                    // Write the data in.
+                    if let Some(mut buffer) = render_queue.write_buffer_with(
+                        buffer,
+                        allocated_range.allocation.offset as u64 * slot_size,
+                        size,
+                    ) {
+                        let slice = &mut buffer.as_mut()[..len];
+                        fill_data(slice);
+                    }
```

Comment:
According to the documentation of `write_buffer_with` it only returns None if the buffer doesn't have enough bytes at the given offset. That seems like an invariant of the code, so we might want to panic or debug_assert instead of silently not copying the mesh?

Reply:
as is, it preserves the old behavior. the method it used to use for writing, `write_buffer`, says "This method fails if `data` overruns the size of `buffer` starting at `offset`."

I digged into the wgpu source, and neither of the two are silent about it, they both print errors to console:
```rs
match self.0.queue_write_buffer(queue_data.id, buffer_data.id, offset, data) {
    Ok(()) => (),
    Err(err) => {
        self.handle_error_nolabel(&queue_data.error_sink, err, "Queue::write_buffer")
    }
}
```

 I think its fine to leave it as is.